tracer/src/Datadog.Tracer.Native/tracer_tokens.cpp

```diff
@@ -199,17 +207,57 @@ HRESULT TracerTokens::EnsureBaseCalltargetTokens()
             bubbleUpExceptionTypeRef = mdTypeRefNil;
         }
     }
+
+    if (callTargetTypeRef != mdTypeRefNil && createRefStructMemberRef == mdMemberRefNil)
+    {
+        COR_SIGNATURE createInstanceSig[32];
+        COR_SIGNATURE* sigBuilder = createInstanceSig;
+        sigBuilder += CorSigCompressData(IMAGE_CEE_CS_CALLCONV_DEFAULT, sigBuilder); // static
+        sigBuilder += CorSigCompressData(2, sigBuilder);                             // arguments count
+        sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_VALUETYPE, sigBuilder);   // return type
+        sigBuilder += CorSigCompressToken(callTargetRefStructTypeRef, sigBuilder);
+        sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_PTR, sigBuilder);           // ptr
+        sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_VOID, sigBuilder);           // void
+        sigBuilder += CorSigCompressElementType(ELEMENT_TYPE_VALUETYPE, sigBuilder);     // RuntimeTypeHandle type
+        sigBuilder += CorSigCompressToken(runtimeTypeHandleRef, sigBuilder);
+
+        auto hr = module_metadata->metadata_emit->DefineMemberRef(
+            callTargetTypeRef, managed_profiler_calltarget_createrefstruct_name.data(), createInstanceSig, sigBuilder - createInstanceSig,
+            &createRefStructMemberRef);
+        if (FAILED(hr))
+        {
+            Logger::Warn("Wrapper CreateRefStruct could not be defined.");
+            return hr;
+        }
+    }
+
     return hr;
 }
 
-int TracerTokens::GetAdditionalLocalsCount()
+int TracerTokens::GetAdditionalLocalsCount(const std::vector<TypeSignature>& methodTypeArguments)
 {
+    int refStructCount = 0;
+    if (enable_by_ref_instrumentation)
+    {
+        const auto module_metadata = GetMetadata();
+        for (int i = 0; i < methodTypeArguments.size(); i++)
```

Comment:
nit: you can you `for-range-based loop` here I think
```
for (auto const& typeArgument : methodTypeArguments)
{
...
}
```

Reply:
done in [bb59282](https://github.com/DataDog/dd-trace-dotnet/pull/5442/commits/bb59282665a0d799431a31a11a6bdd5d9c477a09)